smart-contracts/contracts/interfaces/IPublicLock.sol

```diff
@@ -216,6 +224,17 @@ contract IPublicLock is IERC721Enumerable {
     bytes calldata _data
   ) external payable;
 
+  /**
+   * @notice returns the minimum price paid for a purchase with these params.
+   * @dev this considers any discount from Unlock or the OnKeySold hook.
+   */
+  function purchasePriceFor(
```

Comment:
Maybe we move this to the `LockInspector` instead

Reply:
Hmm but the hook might want to use msg.sender to determine the lock. So not sure we could move it to an extension...